### PR TITLE
Changed UI, and added voting and answering functionality

### PR DESCRIPTION
**Pull request by jessiewon**
**What does this PR do?**
This PR has an updated UI, with the logo and favicon changed with the marketing files and the name changed from Wize-Q to WizeQ. The anonymous checkbox is also automatically checked when asking a question, and the features of adding an answer and submitting votes were added on the front-end.

**How should this be manually tested?**
I used myjson.com to test the UI and GET with the data found in TestQuestions.json, however waiting on the backend before I can formally test all the POST requests.

**What are the relevant tickets?**
https://wizeline.atlassian.net/browse/WIP-214
https://wizeline.atlassian.net/browse/WIP-212
https://wizeline.atlassian.net/browse/WIP-191

**Screenshots (if appropriate)**
screen shot 2017-05-16 at 4 27 54 pm

### CODE REVIEW

File: src/components/List.js [hunk 1]
```diff
@@ -25,25 +35,70 @@ class List extends Component {
...
+  handleAnswerChange(event) {
+    this.setState({ answerValue: event.target.value });
+  }
+
+  submitAnswer() {
+    if (this.state.answerValue === '') {
+      alert('Answer Needed to Post');
+    } else {
+      axios.post('http://51d675a7.ngrok.io/post', qs.stringify({
+        id: this.state.questionClicked,
```
Comment (alan-maldonado) [inline src/components/List.js]: I think with axios you don't need to stringify =)

**Comment by rominavarela** [comment]
Please be clear with your commits. Next time select the chunks of code that belong to "Add voting" and "Add answering" separately. Remember for this you can use visual tools like [SourceTree](https://www.sourcetreeapp.com/).

**Comment by rominavarela** [comment]
Please add some unit tests, Jest was configured in the last sprint [PR](https://github.com/wizeline/wize-q-front/pull/2). Look for writing tests in Jest.

File: src/components/Intro.js [hunk 2]
```diff
@@ -4,7 +4,7 @@ import '../styles/Intro.css';
 const Intro = function () {
   return (
     <div className="Intro">
-      <h1>Welcome to Wize-Q!</h1>
+      <h1>Welcome to WizeQ!</h1>
       <p>
         This is an internal onboarding project to help us better
         answer your questions.
```
Comment (codinronan) [inline src/components/Intro.js]: Not part of the PR I know, but let's change this to not specifically mention onboarding. We could say something like:

`Submit your questions for TGIF and the team meeting, anonymously.`

Meh. I'm not a copywriter :) Maybe ask Bere Beltran or Maribel Carillo for a quick inspiration.

File: src/components/List.js [hunk 3]
```diff
@@ -3,18 +3,28 @@ import React, { Component } from 'react';
...
       searchTerm: '',
+      showModal: false,
+      questionClicked: null,
+      answerValue: '',
     };
+    this.openModal = this.openModal.bind(this);
+    this.closeModal = this.closeModal.bind(this);
+    this.handleAnswerChange = this.handleAnswerChange.bind(this);
+    this.submitAnswer = this.submitAnswer.bind(this);
     this.createList = this.createList.bind(this);
```
Comment (codinronan) [inline src/components/List.js]: Let's get rid of this list of `bind`s, and instead add our member functions as fat arrow functions:

`openModal = (questionId) => {
... function body...
}
```

We should have the appropriate Babel config for this to be transpiled and it makes the code so much cleaner.

File: src/components/List.js [hunk 4]
```diff
@@ -3,18 +3,28 @@ import React, { Component } from 'react';
...
   }
 
   componentDidMount() {
-    axios.get('http://51d675a7.ngrok.io/list')
+    axios.get('https://api.myjson.com/bins/1ce90p')
```
Comment (codinronan) [inline src/components/List.js]: At the very least, define these kinds of constants at the top of the page as `const SOME_URL =...`

So easy to lose them when they're embedded in the code and it'll make them hard to change later.

File: src/components/List.js [hunk 5]
```diff
@@ -25,25 +35,70 @@ class List extends Component {
...
+  submitAnswer() {
+    if (this.state.answerValue === '') {
+      alert('Answer Needed to Post');
+    } else {
+      axios.post('http://51d675a7.ngrok.io/post', qs.stringify({
+        id: this.state.questionClicked,
+        answer: this.state.answerValue,
+      })).then(alert('Answer Submitted'),
+      ).catch(error => console.log(error),
+      );
```
Comment (codinronan) [inline src/components/List.js]: We don't need the extra commas at the end. This is kind of hard to read. Better is:

```
...
.then(() => alert('Answer Submitted'))
.catch((error) => console.warn(error));
```

And yes, start the `.then` on the next line

File: src/components/List.js [hunk 6]
```diff
@@ -25,25 +35,70 @@ class List extends Component {
...
+  }
+
+/* eslint class-methods-use-this: ["error", { "exceptMethods": ["submitVote"] }] */
+  submitVote(voteValue, questionId) {
+    axios.post('http://51d675a7.ngrok.io/post', qs.stringify({
+      id: { questionId },
+      is_positive: { voteValue },
+    })).then(alert('Vote Submitted'),
+    ).catch(error => console.log(error),
+    );
```
Comment (codinronan) [inline src/components/List.js]: Same

File: src/components/List.js [hunk 7]
```diff
@@ -25,25 +35,70 @@ class List extends Component {
...
           <div className="num-votes">
             {question.num_up_vote} <span className="glyphicon glyphicon-thumbs-up up-votes" />
             {question.num_down_vote} <span className="glyphicon glyphicon-thumbs-down down-votes" />
+            Asked by
+            {(() => {
+              if (question.is_anonymous === true) {
+                return ' Anonymous';
+              }
+              return <span> {question.username}</span>;
+            })()}
```
Comment (codinronan) [inline src/components/List.js]: This is great, but hard to read - whenever you do more than trivial operations in JSX, please break it out into a separate function that you then call:

```
{this.whatever()}
```

File: src/components/List.js [hunk 8]
```diff
@@ -16,61 +17,62 @@ class List extends Component {
...
-    this.openModal = this.openModal.bind(this);
-    this.closeModal = this.closeModal.bind(this);
-    this.handleAnswerChange = this.handleAnswerChange.bind(this);
-    this.submitAnswer = this.submitAnswer.bind(this);
-    this.createList = this.createList.bind(this);
   }
 
-  componentDidMount() {
-    axios.get('https://api.myjson.com/bins/1ce90p')
+  componentDidMount = () => {
```
Comment (codinronan) [inline src/components/List.js]: Oops! I should have been more clear, this approach is not needed for the lifecycle functions since Component automatically binds those ones for you :)

File: src/components/List.js [hunk 9]
```diff
@@ -16,61 +17,62 @@ class List extends Component {
...
-      id: { questionId },
-      is_positive: { voteValue },
-    })).then(alert('Vote Submitted'),
-    ).catch(error => console.log(error),
-    );
+  submitVote = (voteValue, questionId) => {
+    axios.post(`${API_URL}/questions`, ({
+      QuestionId: questionId,
+      x_user_id: 27,
+      is_positive: voteValue }))
```
Comment (codinronan) [inline src/components/List.js]:
```
    is_positive: voteValue
}))
```
:)

Sorry to be a nag :)

File: src/components/List.js (hunk 8 above)
Comment (jessiewon) [inline src/components/List.js]: Ohhh haha oops, I'll fix it, thanks Patrick!
